test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -1097,6 +1174,35 @@ public Builder screenCapture() {
             return this;
         }
 
+        /**
+         * Adds a log area below the "Pass", "Fail" buttons.
+         * <p> The log area can be controlled by {@link #log(String)},
+         * {@link #logClear()} and {@link #logSet(String)}.
+         *
+         * @return this builder
+         */
+        public Builder logArea() {
+            this.addLogArea = true;
```

Comment:
```suggestion
            addLogArea = true;
```
I think the usage of `this.` is redundant when there's no naming conflict.

Reply:
I keep it for consistency across builder methods, e.g. `screenCapture` does not need to specify it either.

Reply:
Okay. Most methods of the builder use `this.` prefix when assigning something to a field.

Reply:
How about having a clear log button. Just an idea
Example : 
I have a mouse event test & I am using the logArea to add all the events into it. But when I have to file a bug & I wanted to show only the required log in the log area in the screen shot by clicking on the 'screen capture' button. I have to clear the log in the logarea and repeat the test and show only the required log in the log area. In this scenario if we can provide a 'Clear log' button will be much useful .

Reply:
'Clear log' button calling logClear().

Reply:
@lawrence-andrew I can't see much value…

Capturing logs by a screenshot is not something that you should ever do — copy the messages and paste them as text into a bug report.

However, you've raised a good point: the log messages could / should be copied to System.out too.

If a test fails, the logs will already be available in the test log, and the tester will not need to copy the log to preserve the messages for submitting the bug. The .jtr file from jtreg will contain all the information.

When submitting a bug, you would filter the log and paste the relevant pieces into the bug report.

Reply:
My original intention was to filter the log and paste the relevant pieces into the bug report. But I used screen shot as an example. Thanks

Reply:
> How about having a clear log button.

Maybe later, I don't have a good argument for that yet.

> My original intention was to filter the log and paste the relevant pieces into the bug report. But I used screen shot as an example. Thanks

The text in the log area can be selected with the mouse and copied with `ctrl + c`.


> However, you've raised a good point: the log messages could / should be copied to System.out too.

We could add smth like:

```java
    public static void log(String message) {
        System.out.println("PassFailJFrame: " + message);
        invokeOnEDTUncheckedException(() -> logArea.append(message + "\n"));
    }


    public static void logClear() {
        System.out.println("PassFailJFrame: log cleared");
        invokeOnEDTUncheckedException(() -> logArea.setText(""));
    }

    public static void logSet(String text) {
        System.out.println("PassFailJFrame: log set to:\n" + text);
        invokeOnEDTUncheckedException(() -> logArea.setText(text));
    }
```

Reply:
> > How about having a clear log button.
> 
> Maybe later, I don't have a good argument for that yet.

I can't see a clear use case for the Clear Log button.

I'd rather not add another UI element until we have a real use-case for it. We'll handle it when we face one.

> > My original intention was to filter the log and paste the relevant pieces into the bug report. But I used screen shot as an example. Thanks
> 
> The text in the log area can be selected with the mouse and copied with `ctrl + c`.

The text can be selected by <kbd>Ctrl</kbd>+<kbd>A</kbd>.

If the messages are also logged to `System.out`, the messages will be preserved by the test harness.

> > However, you've raised a good point: the log messages could / should be copied to System.out too.
> 
> We could add smth like:

Yep, that's what I've thinking about.

I'll print an additional blank line `\n` to the log before clearing (or after clearing) and before replacing the text.

Reply:
> I can't see a clear use case for the Clear Log button.
> 
> I'd rather not add another UI element until we have a real use-case for it. We'll handle it when we face one.

I agree, adding in incremental steps, as & when we encounter a use case is better. The log area feature, as it is now looks good. As logClear() & logSet() are already in place any future UI extension around it can be done easily.

Reply:
> I'll print an additional blank line \n to the log before clearing (or after clearing) and before replacing the text.

Updated.